contrib/gitian-build.sh

```diff
@@ -1,406 +1,367 @@
-	-j)
-	    if [ -n "$2" ]
-	    then
-		proc=$2
-		shift
-	    else
-		echo 'Error: "-j" requires an argument'
-		exit 1
-	    fi
-	    ;;
-	# Memory to allocate
-	-m)
-	    if [ -n "$2" ]
-	    then
-		mem=$2
-		shift
-	    else
-		echo 'Error: "-m" requires an argument'
-		exit 1
-	    fi
-	    ;;
-	# URL
-	-u)
-	    if [ -n "$2" ]
-	    then
-		url=$2
-		shift
-	    else
-		echo 'Error: "-u" requires an argument'
-		exit 1
-	    fi
-	    ;;
-        # kvm
-        --kvm)
-            lxc=false
+        -o | --os)
+            if [ -n "$2" ]; then
+                DESCRIPTORS=()
+                SIGN_DESCRIPTORS=()
+                if [[ "$2" == *"l"* ]]; then
+                    DESCRIPTORS+=('linux')
+                fi
+                if [[ "$2" == *"w"* ]]; then
+                    DESCRIPTORS+=('win')
+                    SIGN_DESCRIPTORS+=('win-signer')
+                fi
+                if [[ "$2" == *"x"* ]]; then
+                    DESCRIPTORS+=('osx')
+                    SIGN_DESCRIPTORS+=('osx-signer')
+                fi
+                shift
+            else
+                echo 'Error: "--os" specify os: l (linux), w (windows), or x (Mac OSX)'
+                exit 1
+            fi
+            ;;
+        # Help message
+        -h | --help)
+            echo "$usage"
+            exit 0
+            ;;
+        # Commit or branch
+        -c | --commit)
+            commit=true
+            ;;
+        # Number of Processes
+        -j)
+            if [ -n "$2" ]; then
+                proc=$2
+                shift
+            else
+                echo 'Error: "-j" requires an argument'
+                exit 1
+            fi
+            ;;
+        # Memory to allocate
+        -m)
+            if [ -n "$2" ]; then
+                mem=$2
+                shift
+            else
+                echo 'Error: "-m" requires an argument'
+                exit 1
+            fi
+            ;;
+        # lxc
+        --lxc)
+            USE_LXC=1
             ;;
-        # Detach sign
-        --detach-sign)
-            signProg="true"
-            commitFiles=false
+        # docker
+        --docker)
+            USE_DOCKER=1
             ;;
-        # Commit files
-        --no-commit)
-            commitFiles=false
+        # URL
+        -u)
+            if [ -n "$2" ]; then
+                url=$2
+                shift
+            else
+                echo 'Error: "-u" requires an argument'
+                exit 1
+            fi
             ;;
         # Setup
         --setup)
             setup=true
             ;;
-	*)               # Default case: If no more options then break out of the loop.
-             break
+        --test)
+            test=true
+            ;;
+        *) # Default case: If no more options then break out of the loop.
+            break ;;
     esac
     shift
 done
 
-# Set up LXC
-if [[ $lxc = true ]]
-then
-    export USE_LXC=1
-    export LXC_BRIDGE=br0
```

Comment:
**IMPORTANT:** This should still be set for LXC build to work, recommend doing that above when processing `--lxc`, but do an optional setting, so that it can be overridden (on systems where I have multiple virtualization solutions, this is nearly never `br0`). Alternatively, document it in the guide that this must ALWAYS be set or add a parameter.

Reply:
Do you want to go with something like that in option parsing :
```bash
--lxc)
    USE_LXC=1
    export LXC_BRIDGE=${LXC_BRIDGE:-br0}
    ;;
```
Defining `br0` as default value, and letting the user override it if needed ?